hydra-node/test/Hydra/Chain/Direct/HandlersSpec.hs

```diff
@@ -70,33 +77,61 @@ import Test.QuickCheck.Monadic (
   assert,
   monadicIO,
   monitor,
-  pick,
   run,
   stop,
  )
 import qualified Prelude
 
+genTimeHandleWithSlotInsideHorizon :: Gen (TimeHandle, SlotNo)
+genTimeHandleWithSlotInsideHorizon = do
+  (systemStart, eraHistory, horizonSlot, currentTime) <- genTimeParams
+  let timeHandle = mkTimeHandle currentTime systemStart eraHistory
+  pure (timeHandle, horizonSlot - 1)
+
+genTimeHandleWithSlotPastHorizon :: Gen (TimeHandle, SlotNo)
+genTimeHandleWithSlotPastHorizon = do
+  (systemStart, eraHistory, horizonSlot, currentTime) <- genTimeParams
+  let timeHandle = mkTimeHandle currentTime systemStart eraHistory
+  pure (timeHandle, horizonSlot + 1)
```

Comment:
Should we test with variable number of slots before/past horizon? Maybe we can use numbers up to 3k to see if the conversion works for those and then see that the failure happens if we go beyond 3k blocks?

Reply:
The `horizonSlot` is already the boundary. The 3k slots would already lead to the `horizon` being where it is.

However, we should be able to generate various slots inside or past the horizon easily now:
```
inside <- choose (0, horizonSlot)
past <- (horizonSlot +) <$> arbitrary
```

Reply:
oh right:
```
 safeZone = 3 * 2160 / 0.05
 horizonSlot = SlotNo $ truncate $ uptimeSeconds + safeZone
```